### PR TITLE
Add support for strikethrough formatting on text.

### PR DESCRIPTION
**Pull request by smmalis37**
Made by basically just copying underline. Just like underline this does not have any support for formatting the strikethrough itself, although the underlying apis support it. ~~I do not have access to a mac to test the coregraphics implementation (nor have I ever written mac code before)~~, however the d2d implementation appears to work on my Win 10 box. This will need a snapshots update.

### CODE REVIEW

**Comment by smmalis37** [comment]
Whoops, I thought one of the files I was editing was piets, its actually core-graphics (vscode just lets you edit source of downloaded crates, fun). So that implementation won't work at the moment. Investigating what's needed.

**Comment by smmalis37** [comment]
Blocked on https://github.com/servo/core-foundation-rs/pull/409

**Comment by smmalis37** [comment]
From Zulip:
> Raph Levien: The best path forward might be to stub out the Apple implementation and open an issue to add it.

I'm fine with that if the maintainers are.

File: piet/src/text.rs
Comment (cmyr) [inline piet/src/text.rs:126]:
```suggestion
    /// Strikethrough.
```

**Review by cmyr — APPROVED** [review]
Okay, I'm happy to have this go in as-is; at some point I'd like to think more about a better API for "text decorations" but this is totally reasonable for now.

We'll need an issue that this needs implementation on macOS and cairo.

**Pull request merged**
